custom-completions/git/git-completions.nu

```diff
@@ -24,37 +46,41 @@ def "nu-complete git commits" [] {
   ^git rev-list --all --remotes --pretty=oneline | lines | parse "{value} {description}"
 }
 
+# This includes local branches, remote branches with remote prefix and commits.
+# Accepted at:
+# `git checkout <value>`
+# `git rebase --onto <value> <value>`
 def "nu-complete git branches and commits" [] {
-  nu-complete git switchable branches
+  nu-complete git local and remote branches
   | parse "{value}"
   | insert description Branch
   | append (nu-complete git commits)
 }
 
 # Check out git branches and files
 export extern "git checkout" [
-  ...targets: string@"nu-complete git switchable branches"   # name of the branch or files to checkout
```

Comment:
What do we do here? Add yet another completion command or concatenate the output of two and make it unique?
The below commands are all valid
- `git checkout local-branch`
- `git checkout remote-branch-without-prefix`
- `git checkout upstream/remote-branch-without-prefix`
- `git checkout 234commit456hash`

Reply:
Maybe I'm wrong here, but I'm thinking we'd concatenation and unique the list.

Reply:
OK that would work :+1: One further question: Should the completion suggestions be a flat list (as in the current version of `git switch`) or a table with `value`/`description` pairs? The flat list makes better use of limited terminal space, but especially commit suggestions may be useless without any description alongside.

Reply:
Could you maybe use a parameter to allow one to choose --verbose to show the value/description version?

Reply:
Hm but how exactly would that work? A parameter to using the completion, or a parameter in the `git` command? :thinking: